pkg/cloudprovider/azure.go

```diff
@@ -170,9 +170,9 @@ func (a *Azure) AssignPrivateIP(ip net.IP, node *corev1.Node) error {
 	// UserDefinedRouting, which doesn't impose such constraints on secondary IPs.
 	loadBalancerBackendAddressPoolsArgument := (*networkInterface.IPConfigurations)[0].LoadBalancerBackendAddressPools
 	var attachedOutboundRule *network.SubResource
-	if (*networkInterface.IPConfigurations)[0].LoadBalancerBackendAddressPools != nil {
-	OuterLoop:
-		for _, ipconfig := range *networkInterface.IPConfigurations {
+OuterLoop:
+	for i, ipconfig := range *networkInterface.IPConfigurations {
+		if (*networkInterface.IPConfigurations)[i].LoadBalancerBackendAddressPools != nil {
```

Comment:
is there a reason why you're not using `ipconfig` here?

Reply:
Sorry, my bad. I should have used ipconfig instead. I will correct this and push soon.
Thanks @ricky-rav for pointing this out.